src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -158,6 +160,58 @@ private void testMapAllTypes(boolean registerFactory) throws SQLException {
     }
   }
 
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapJsonToMap() throws SQLException {
+    Connection connection = init();
+    Statement statement = connection.createStatement();
+    ResultSet resultSet =
+        statement.executeQuery("SELECT OBJECT_CONSTRUCT('string','a','string2','s')");
+    resultSet.next();
+    Map map = resultSet.getObject(1, Map.class);
+    assertEquals("a", map.get("string"));
+    assertEquals("s", map.get("string2"));
+    statement.close();
+    connection.close();
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testReturnAsMap() throws SQLException {
+    SnowflakeObjectTypeFactories.register(SimpleClass.class, SimpleClass::new);
+    Connection connection = init();
+    Statement statement = connection.createStatement();
+    ResultSet resultSet =
+        statement.executeQuery(
+            "select {'x':{'string':'one'},'y':{'string':'two'},'z':{'string':'three'}}::MAP(VARCHAR, OBJECT(string VARCHAR));");
+    resultSet.next();
+    Map<String, SimpleClass> map =
+        resultSet.unwrap(SnowflakeBaseResultSet.class).getMap(1, SimpleClass.class);
+    assertEquals(map.get("x").getString(), "one");
+    assertEquals(map.get("y").getString(), "two");
+    assertEquals(map.get("z").getString(), "three");
+    statement.close();
+    connection.close();
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testReturnAsList() throws SQLException {
```

Comment:
are we able to parameterize the added tests? the tests here a focused on ResultSet and processing it so we can create some function taking lambda to run in the tests?

Reply:
Fixed